app/src/test/java/com/example/bottomnav/DataTest.java

```diff
@@ -0,0 +1,24 @@
+package com.example.bottomnav;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class DataTest {
+    @Test
+    public void randomData() {
+        String data = "t056111";
+        CAN_Data newData = CAN_Data.decode(data);
+        System.out.println(newData.getData());
+        assertEquals( 0x11 , newData.getData()[0]);
+        assertEquals( 86 , newData.getId());
+    }
+    @Test
+    public void BIGData() {
+        String data = "t0568111111111111111A";
+        CAN_Data newData = CAN_Data.decode(data);
+        System.out.println(newData.getData());
+        assertEquals( 0x11, newData.getData()[0]);
```

Comment:
I'd recommend checking all the payload data.
And all the parsed data - including ID.